### PR TITLE
Add MEI export for multiRest

### PR DESCRIPTION
**Pull request by rettinghaus**
@lpugin please have a look

### CODE REVIEW

**Comment by lpugin** [comment]
This is what you need to add / change:

First of all the features implemented have import and export with a round trip test, so it should be the same here. You need to implement the importer, add a test case and make sure the round-trip test passes.

Most MEI elements are set in the MeiConverter class. That way we can have import / export in the same place as much as possible. You should do the same here.

**Comment by lpugin** [comment]
It should be converted into a draft.

File: src/importexport/mei/internal/meiexporter.cpp [hunk 1]
```diff
@@ -893,6 +893,17 @@ bool MeiExporter::writeLayer(track_idx_t track, const Staff* staff, const Measur
     meiLayer.SetN(static_cast<int>(track2voice(track) + 1));
     meiLayer.Write(m_currentNode, this->getLayerXmlId());
 
+    const Measure* mmR1 = measure->coveringMMRestOrThis();
+    if (mmR1->mmRestFirst()) {
```
Comment (lpugin) [inline src/importexport/mei/internal/meiexporter.cpp:897]: You cannot call `mmRestFirst` without checking the validity of `mmR1`

File: src/importexport/mei/internal/meiexporter.cpp [hunk 2]
```diff
@@ -893,6 +893,17 @@ bool MeiExporter::writeLayer(track_idx_t track, const Staff* staff, const Measur
...
+        pugi::xml_node multiRestNode = m_currentNode.append_child();
+        libmei::MultiRest meiMultiRest;
+        meiMultiRest.SetNum(mmR1->mmRestCount());
+        std::string xmlId = this->getXmlIdFor(mmR1, 'm');
+        meiMultiRest.Write(multiRestNode, xmlId);
+        m_currentNode = m_currentNode.parent();
+        return true;
+    }
+
```
Comment (lpugin) [inline src/importexport/mei/internal/meiexporter.cpp:906]: Make an `else` statement here and do not return in the `if` statement above. That way we do not start changing the value of `m_currentNode` at different places, which is quickly error prone.

File: src/importexport/mei/internal/meiexporter.cpp [hunk 3]
```diff
@@ -893,6 +893,17 @@ bool MeiExporter::writeLayer(track_idx_t track, const Staff* staff, const Measur
     meiLayer.SetN(static_cast<int>(track2voice(track) + 1));
     meiLayer.Write(m_currentNode, this->getLayerXmlId());
 
+    const Measure* mmR1 = measure->coveringMMRestOrThis();
+    if (mmR1->mmRestFirst()) {
+        pugi::xml_node multiRestNode = m_currentNode.append_child();
+        libmei::MultiRest meiMultiRest;
+        meiMultiRest.SetNum(mmR1->mmRestCount());
+        std::string xmlId = this->getXmlIdFor(mmR1, 'm');
+        meiMultiRest.Write(multiRestNode, xmlId);
```
Comment (lpugin) [inline src/importexport/mei/internal/meiexporter.cpp:902]: To be more inline with the rest of the exporter structure, make a `writeMultiRest` method for this.

**Review by lpugin — CHANGES_REQUESTED** [review]

File: src/importexport/mei/internal/meiexporter.cpp [hunk 4]
```diff
@@ -893,6 +893,11 @@ bool MeiExporter::writeLayer(track_idx_t track, const Staff* staff, const Measur
     meiLayer.SetN(static_cast<int>(track2voice(track) + 1));
     meiLayer.Write(m_currentNode, this->getLayerXmlId());
 
+    const Measure* mmR1 = measure->coveringMMRestOrThis();
+    if (measure != mmR1 && mmR1->mmRestFirst()) {
```
Comment (lpugin) [inline src/importexport/mei/internal/meiexporter.cpp:897]: you are still not checking the validity of `mmR1`

File: src/importexport/mei/internal/meiexporter.cpp [hunk 5]
```diff
@@ -1245,6 +1250,25 @@ bool MeiExporter::writeNote(const Note* note, const Chord* chord, const Staff* s
...
+
+    pugi::xml_node multiRestNode = m_currentNode.append_child();
+    libmei::MultiRest meiMultiRest;
+    meiMultiRest.SetNum(mmRest->mmRestCount());
+    std::string xmlId = this->getXmlIdFor(mmRest, 'm');
+    meiMultiRest.Write(multiRestNode, xmlId);
+
+    return true;
+}
+
```
Comment (lpugin) [inline src/importexport/mei/internal/meiexporter.cpp:1271]: Methods are kept ordered alphabetically.

**Review by lpugin — CHANGES_REQUESTED** [review]

**Comment by lpugin** [comment]
@rettinghaus I think it would be simpler for such PRs to be made to https://github.com/rism-digital/MuseScore and we can the make a PR here once it is ready. @cbjeukendrup what do you think?

**Comment by cbjeukendrup** [comment]
@lpugin I've no objections against that. Feel free to do whatever is most convenient for you :)

**Comment by lpugin** [comment]
Yes, for cases like this one it would make it easier for me to review and create less noise on the main repo with things that are WIP steps in the MEI support.

**Comment by rettinghaus** [comment]
Ok

**Pull request closed without merging**